Assets/Scripts/Game/UI/PopupSystem.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PopupSystem : MonoBehaviour
+{
+    [SerializeField] private Transform canvas;
+    [SerializeField] private PopupWindow popupPrefab;
+
+    public GameObject CreatePopup(string title, GameObject content)
+    {
+        PopupWindow window = Instantiate(popupPrefab, canvas);
+        window.Title = title;
+
+        Instantiate(content, window.Content);
+        return window.gameObject;
+    }
+
+    public GameObject CreatePopup(string title, out Transform content)
+    {
+        PopupWindow window = Instantiate(popupPrefab, canvas);
+        window.Title = title;
+        content = window.Content.transform;
+        return window.gameObject;
+    }
```

Comment:
If you want to have persistent content instead of re-instantiating it every time a popup opens (instantiate is slow), having the content transform returned is really handy. You can just do something like
```csharp
private Transform popupContent; //This gets initialized at some point

//Later...
PopupSystem.CreatePopup("Popup Name", out Transform content);
popupContent.parent = content;
```

Reply:
Gotcha... is there any reason we aren't using this approach in the QuestSystem then? Seems like windows like the notes panel are a perfect target for this approach.

Reply:
I'm planning on implementing it like that soon, I just wanted to get a PR in to make sure everything was headed in the right direction :)